addons/html_builder/static/src/interactions/google_map.edit.js

```diff
@@ -3,83 +3,78 @@
-            } else {
-                return false;
+    /**
+     * Get the stored API key if any (or open a dialog to ask the user for one),
+     * load and configure the Google Maps API.
+     *
+     * @param {boolean} [forceReconfigure=false]
+     * @returns {Promise<void>}
+     */
+    async loadGoogleMaps(forceReconfigure = false) {
+        /** @type {string | undefined} */
+        const apiKey = await this.websiteMapService.getGMapAPIKey(true);
+        const apiKeyValidation = await this.websiteMapService.validateGMapApiKey(apiKey);
+        const shouldReconfigure = forceReconfigure || !apiKeyValidation.isValid;
+        let didReconfigure = false;
+        if (shouldReconfigure) {
+            didReconfigure = await this.websiteEditService.callShared(
+                "googleMapsOption",
+                "configureGMapsAPI",
+                apiKey
+            );
+            if (!didReconfigure) {
+                this.websiteEditService.callShared("remove", "removeElement", this.el);
             }
         }
-
-        /**
-         * Load the Google Maps API from the Google Map Service.
-         * This method is set apart so it can be overridden for testing.
-         *
-         * @param {boolean} [shouldRefetch]
-         * @returns {Promise<string|undefined>} A promise that resolves to an API
-         *                                      key if found.
-         */
-        async loadGoogleMapsAPIFromService(shouldRefetch) {
-            const apiKey = await this.websiteMapService.loadGMapAPI(true, shouldRefetch);
-            this.websiteEditService.callShared("googleMapsOption", "shouldNotRefetchApiKey");
-            return !!apiKey;
+        if (!shouldReconfigure || didReconfigure) {
+            const shouldRefetch = this.websiteEditService.callShared(
+                "googleMapsOption",
+                "shouldRefetchApiKey"
+            );
+            return !!(await this.loadGoogleMapsAPIFromService(shouldRefetch || didReconfigure));
+        } else {
+            return false;
         }
-    };
+    }
+
+    /**
+     * Load the Google Maps API from the Google Map Service.
+     * This method is set apart so it can be overridden for testing.
+     *
+     * @param {boolean} [shouldRefetch]
+     * @returns {Promise<string|undefined>} A promise that resolves to an API
+     *                                      key if found.
+     */
+    async loadGoogleMapsAPIFromService(shouldRefetch) {
+        const apiKey = await this.websiteMapService.loadGMapAPI(true, shouldRefetch);
+        this.websiteEditService.callShared("googleMapsOption", "shouldNotRefetchApiKey");
+        return !!apiKey;
+    }
+}
 
-registry.category("public.interactions.edit").add("website.google_map", {
-    Interaction: GoogleMap,
-    mixin: GoogleMapsEdit,
+registry.category("public.interactions.edit").add("html_builder.google_map", {
+    Interaction: GoogleMapsEdit,
```

Comment:
Ok, the "edit" one is not actual overloads, so it can be distinct... but then the "visitor" one also needs to be registered in edit mode, so that it runs during edition.